tests/unit/track/test_track.py

```diff
@@ -192,3 +212,112 @@ def test_filter_trajectories(self, mocker):
+
+        with Dataset(file_path, "w", format="NETCDF4") as nc:
+            # Dimensions
+            nc.createDimension("lon", self.nx)
+            nc.createDimension("lat", self.ny)
+            nc.createDimension("time", None)  # unlimited
+
+            # Time data
+            time = nc.createVariable("time", "f4", ("time",))
+            time[:] = [0, 0.5, 1, 1.5, 2]
+            time.standard_name = "time"
+            time.long_name = "time"
+            time.units = "days since 1950-01-01"
+            time.calendar = "360_day"
+            time.axis = "T"
+
+        return str(file_path)
+
+    def test_trajectories(self, mocker, tmp_path):
+        """Check trajectories correctly reads in the output netcdf file."""
+        # Create the temporary input and output files
+        self._create_nc_output_file(tmp_path)
+        input_file = self._create_nc_input_file(tmp_path)
+
+        # Set up the tracker pointing to the created files
+        params = {"base_dir": str(tmp_path), "input_file": input_file}
+        tracker = self._setup_tracker(mocker, mock_input_file=False, params_dict=params)
+
+        # Get the trajectories
+        trajectories = tracker.trajectories()
+
+        # Check the trajectories
+        t1 = trajectories[0]
+        assert t1.trajectory_id == 0
+        assert t1.observations == 2
+        assert t1.calendar == "360_day"
+        assert_array_equal(t1.data["lon"], [0, 1])
+        assert_array_equal(t1.data["lat"], [0, -1])
+        assert_array_equal(t1.data["intensity"], [0, 10])
+        times1 = cftime.num2date(
+            [0, 0.5], units="days since 1950-01-01", calendar="360_day"
+        )
+        assert_array_equal(t1.data["timestamp"], times1)
+
+        t2 = trajectories[1]
+        assert t2.trajectory_id == 1
+        assert t2.observations == 3
+        assert t2.calendar == "360_day"
+        assert_array_equal(t2.data["lon"], [2, 3, 4])
+        assert_array_equal(t2.data["lat"], [-2, -3, -4])
+        assert_array_equal(t2.data["intensity"], [20, 30, 40])
+        times2 = cftime.num2date(
+            [0, 0.5, 1], units="days since 1950-01-01", calendar="360_day"
+        )
+        assert_array_equal(t2.data["timestamp"], times2)
+
+    def test_trajectories_failure(self, mocker, tmp_path):
+        """Check trajectories fails as expected."""
+        # No output file
+        tracker = self._setup_tracker(mocker)
+        mocker.stopall()
+        with pytest.raises(
+            FileNotFoundError, match="TRACK output trajectory file does not exist"
+        ):
+            tracker.trajectories()
+
+        # No input file (create the output file)
+        self._create_nc_output_file(tmp_path)
+        params = {"base_dir": str(tmp_path)}
+        tracker = self._setup_tracker(mocker, params_dict=params)
+        mocker.stopall()
+        with pytest.raises(FileNotFoundError, match="Input file does not exist"):
+            tracker.trajectories()
```

Comment:
I'd suggest breaking this up into "Unit" tests.
Otherwise you could fail the first part without testing the second.